src/api/lib/backend/api/build_results/binaries.rb

```diff
@@ -31,6 +31,55 @@ def self.build_dependency_info(project, package, repository, architecture)
           path += "?package=#{CGI.escape(package)}&view=pkgnames"
           Backend::Connection.get(path).body.force_encoding("UTF-8")
         end
+
+        # Returns the available binaries for the project
+        def self.available_in_project(project)
+          path = "/build/#{CGI.escape(project)}/_availablebinaries"
+          transform_binary_packages_response(Backend::Connection.get(path).body.force_encoding("UTF-8"))
+        end
+
+        # Returns the available binaries for the urls given from the project
+        def self.available_in_urls(project, urls)
+          return {} if urls.empty?
+          path = "/build/#{CGI.escape(project)}/_availablebinaries"
+          query = []
+          urls.each do |url|
+            query << "url=#{CGI.escape(url)}"
+          end
+          path += "?#{query.join('&')}"
+          transform_binary_packages_response(Backend::Connection.get(path).body.force_encoding("UTF-8"))
+        end
+
+        # Returns the available binaries for the obs_repositories given from the project
+        def self.available_in_repositories(project, repositories)
+          return {} if repositories.empty?
+          path = "/build/#{CGI.escape(project)}/_availablebinaries"
+          query = []
+          repositories.each do |repository|
+            query << "path=#{CGI.escape(repository)}"
+          end
+          path += "?#{query.join('&')}"
```

Comment:
You can simplify it to 
```
path += "?#{urls.to_query('url')}"
```

Reply:
In this case is "path", I will change it accordingly

Reply:
Right! 
However, as I wrote in my other comment! You can combine path and url call!

Reply:
I can't use `urls.to_query('url')` it generates `url[]=x1&url[]=x2` and backend doesn't know nothing about `url[]` params 😢